kale/pipeline/base_trainer.py

```diff
@@ -0,0 +1,168 @@
+# =============================================================================
+# Author: Xianyuan Liu, xianyuan.liu@outlook.com
+#         Haolin Wang, LWang0101@outlook.com
+# =============================================================================
+
+"""Classification of data
```

Comment:
Make the above be more descriptive considering that this will be a more user-facing API that will be reused, learning from the docs from kale.pipeline.domain_adapter as well. And provide references if there are.

Reply:
@XianyuanLiu 
Does this apply to none NN models as well? If yes fine. If not, shall we call it base_nn_trainer?

Reply:
More details are added to the documentation. Renamed them to `base_nn_trainer` and `BaseNNTrainer` for better clarity. 

Non-NN models may follow a similar workflow and inherit some parts of this trainer, but they are not totally the same. So "nn" is added.